website/blog/2024-02-01-making-your-project-multilingual.md

```diff
@@ -0,0 +1,110 @@
+{__('Icon position', 'project-name')}
+```
+
+Alternative functions you can use are:
+- _n
+- _nx
+- _x
+
+You can refer to the [block editor handbook](https://developer.wordpress.org/block-editor/reference-guides/packages/packages-i18n/) for more information on these functions.
+
+## The I18n class
+The easiest way to add I18n support to your project is by using our WP CLI boilerplate command:
+```bash
+wp boilerplate create i18n
+```
+
+This command generated a new class inside the `src/I18n` folder. This class instructs WordPress to look for translations in `src/I18n/languages` with the textdomain defined as your project name. The next step is generating .po and .mo files that are used for translation.
+
+## Generating .pot file
+You can create a .pot file by using WP CLI. Run the following command in your project root:
+```bash
+wp i18n make-pot
+```
+
+Alternatively, you can use [Poedit](https://poedit.net/) to generate the .pot file and generate translations from it later.
+
+## Translating with Poedit
+Once you have the .pot file, you can use Poedit to generate .po and .mo files that are used for translating hardcoded strings in your project. When generating the .po and .mo files, you can choose for which locale you're creating the translation. For example, if you are creating a translation for the German language, your files will be named de_DE.po and de_DE.mo.
+
+After generating the files, you should go to **Translation -> Properties** and go to **Sources Paths** tab. For the Base path, you should set your theme folder. In Excluded paths, you can add folders like `node_modules`, `vendor`, and `public`.
+
+In the **Sources keywords** tab you can set additional functions which you use in your project for translations. Some of the most commonly used include:
+- _e
+- __
+- esc_html__
+- esc_html_e
+- esc_attr__
+- esc_html_x
+- _n
+
+> If you're missing a string in your .po file, be sure to check which function is used for translation for that string and if that function is added to Sources keywords.
+
+After updating the settings, click on the Update from source code option to get the updated list of strings to translate.
+
+The translation process is simple. The left column represents the source text, and the right column the translation. When you have finished translating the strings, copy the .po and .mo files in the `src/I18n/languages` folder.
+
+## JS translations
+The process of translating strings in JS has some extra steps. In order to translate the strings in JS (e.g. block editor), you will have to generate a .json file with translations. To do this, navigate to your `src/I18n/languages` folder and use the following WP-CLI command:
+```bash
+wp i18n make-json <po-file> --no-purge
+```
+
+This will generate a .json file for each JS file you have. This command extracts the strings from your .po files, so you'll already have the translations added. The `--no-purge` flag is used to keep the existing translations in your .po file.
```

Comment:
```suggestion
This will generate a `.json` file for each JS file present. The strings are extracted from `.po` files, so you'll already have the translations added. The `--no-purge` flag is used to keep the existing translations in the `.po` file.
```